appsec/src/extension/commands/request_shutdown.c

```diff
@@ -189,3 +261,355 @@ static void _pack_headers_no_cookies_map(
     ZEND_HASH_FOREACH_END();
     mpack_finish_map(w);
 }
+
+static const char *nullable _header_content_type_zend_array(
+    const zend_array *nonnull hl, size_t *nonnull len)
+{
+
+    zend_string *key;
+    zval *val;
+    zend_long idx;
+    ZEND_HASH_FOREACH_KEY_VAL((zend_array *)hl, idx, key, val)
+    {
+        (void)idx;
+        if (!key) {
+            continue;
+        }
+        if (zend_string_equals_literal_ci(key, "content-type")) {
+            if (Z_TYPE_P(val) == IS_STRING) {
+                *len = Z_STRLEN_P(val);
+                return Z_STRVAL_P(val);
+            }
+            if (Z_TYPE_P(val) == IS_ARRAY) {
+                zend_array *arr = Z_ARR_P(val);
+                HashPosition pos;
+                zend_hash_internal_pointer_end_ex(arr, &pos);
```

Comment:
When is the content type an array and is the actual type always the last element?

Reply:
It always be an array actually. In this case, the headers come from a "user request" (i.e., not the PHP lifecycle), and notify_commit should be given a map {header_name: [value1, ...], ...}.

The last element is taken, because while the rfc 2616 specifies that there should only be one value for content-type (it says only headers defined as comma separated values may have more than one value), I read somewhere that in the wild it's the last value that wins.